lib/module.js

```diff
@@ -77,23 +87,14 @@ module.exports = function sentry(moduleOptions) {
         Sentry.captureException(error)
       }))
     })
-  } else {
-    // This is for backwards compatibility (NuxtJS pre-hooks)
-    this.nuxt.plugin('renderer', (renderer) => {
-      renderer.app.use(Sentry.Handlers.requestHandler())
+  }
 
-      // Grab Nuxt's original error middleware and overwrite it with our own
-      const nuxtErrorMiddleware = renderer.errorMiddleware
-      renderer.errorMiddleware = (err, req, res, next) => {
-        // Log the error
-        Sentry.withScope((scope) => {
-          scope.setExtra('req', req)
-          Sentry.captureException(err)
-        })
+  // Enable publishing of sourcemaps
+  this.extendBuild((config, { isClient, isDev }) => {
+    if (!options.publishRelease || isDev) return
+    if (isClient) config.devtool = '#sourcemap'
```

Comment:
Logic here was not, and still is not, correct IMO. This code here extends both client and server builds with the plugin and yet it only enables generation of source maps for client build. So this code will upload `dist` folder once on building client and then again on building server (without server source maps as those weren't generated).

So I think the proper logic here would be to either:
a) return early on `!isClient` so that we don't upload `dist/server` at all
or
b) always generate source maps and add  `WebpackPlugin` only for `!isClient` (as server bundles are generated second).

For me, I've actually decided to only upload client stuff here as I've enabled sourcemaps for the server in `nuxt.config.js` and I include those in deployed app and that seems to give proper stack traces already without uploading code to Sentry. It's possible that I'm missing out on something by not uploading to Sentry but I'm not sure what.
Given the above, I think I would appriciate a way to configure which source maps to upload to sentry (only client or both).

Reply:
Server maps generated by default as i see, so on second time it will upload both client and server. Also there a modern build so it can be three times :)

Reply:
@rchl i have added some logic to upload only once and allow disabling uploading server or client sources, although logic might be off